newhelm/main.py

```diff
@@ -59,6 +59,29 @@ def list_tests() -> None:
         click.echo(f"Description: {metadata.description}")
         click.echo()
 
+@newhelm_cli.command()
+def list_suts() -> None:
+    """List details about all registered SUTs."""
+    secrets = load_secrets_from_config()
+    for sut, sut_entry in SUTS.items():
+        try:
+            sut_obj = sut_entry.make_instance(secrets=secrets)
+        except MissingSecretValues as e:
+            display_header(
+                f"Cannot display SUT {sut} because it requires the following secrets:"
+            )
+            for secret in e.descriptions:
+                click.echo(secret)
+            click.echo()
+            continue
+        
+        display_header(sut)
+        click.echo(f"class={sut_obj.__class__.__name__}")
+        click.echo(f"args={sut_obj._initialization_record.args}") # Consider create getter function for `_initialization_record`
```

Comment:
> Consider create getter function for `_initialization_record`

You can use [get_initialization_record](https://github.com/mlcommons/newhelm/blob/bfba11f6383f9e7fdd0b114f4b896cc0be5df5e1/newhelm/record_init.py#L48C5-L48C30), although as mentioned in my other comment getting this data directly from FactorEntry is probably cleaner.

Reply:
@brianwgoldman Hello, you explained it very well again, thank you for this and for your feedback. I tried to understand and apply it as accurately as I could. If I make a mistake again, I'm waiting for your feedback!

Sample printouts for missing secret:
```shell
$ poetry run python newhelm/main.py list-suts

llama-2-7b
class=TogetherCompletionsSUT
args=('togethercomputer/llama-2-7b', <newhelm.secret_values.InjectSecret object at 0x0000021F76C56740>)
kwargs={}
Missing required secrets:
scope='together' key='api_key', instructions='See https://api.together.xyz/settings/api-keys'

gpt-3.5-turbo
class=OpenAIChat
args=('gpt-3.5-turbo', <newhelm.secret_values.InjectSecret object at 0x0000021F772FA3B0>, <newhelm.secret_values.InjectSecret object at 0x0000021F772FAF20>)
kwargs={}
Missing required secrets:
scope='openai' key='api_key', instructions='See https://platform.openai.com/api-keys'
```

Sample outputs when the required secret conditions are met:
```shell
$ poetry run python newhelm/main.py list-suts

demo_yes_no
class=DemoYesNoSUT
args=()
kwargs={}
Uses required secrets:
[]

gpt2
class=HuggingFaceSUT
args=('gpt2', <newhelm.secret_values.InjectSecret object at 0x0000021F0D09BCD0>)
kwargs={}
Uses required secrets:
[HuggingFaceToken()]

gpt-3.5-turbo
class=OpenAIChat
args=('gpt-3.5-turbo', <newhelm.secret_values.InjectSecret object at 0x000002168868A350>, <newhelm.secret_values.InjectSecret object at 0x000002168868AEC0>)
kwargs={}
Uses required secrets:
[OpenAIApiKey(), OpenAIOrgId()]
```